packages/backend/src/open_payments/auth/middleware.ts

```diff
@@ -2,6 +2,87 @@ import { AccessType, AccessAction } from './grant'
 import { PaymentPointerContext } from '../../app'
 import { Transaction } from 'objection'
 import { GrantReference } from '../grantReference/model'
+import { createVerifier, httpis, RequestLike } from 'http-message-signatures'
+import { KeyObject } from 'crypto'
+import { Request as KoaRequest } from 'koa'
+import { ClientKeys } from '../../clientKeys/model'
+import { JWKWithRequired } from 'auth'
+
+// Creates a RequestLike object for the http-message-signatures library input
+function requestLike(request: KoaRequest): RequestLike {
+  return {
+    method: request.method,
+    headers: request.headers,
+    url: request.url
+  }
+}
+
+function parseJwkKeyType(jwk: JWKWithRequired): KeyType {
+  if (jwk.kty === 'private' || jwk.kty === 'public' || jwk.kty === 'secret') {
+    return jwk.kty
+  } else {
+    throw new Error('invalid key type')
```

Comment:
The OpenAPI spec only allows `OKP`
https://github.com/interledger/open-payments/blob/s2-split-auth-spec/openapi-specs/shared/schemas.yaml#L188